### ISSUE TITLE
Option to Show Running State in Toast Notification or Statusbar

### ISSUE THREAD

**Issue by sheldonhull** [PowerShell/vscode-powershell#1799]
**Summary of the new feature**

As a user running powershell code, I want a visual indicator that the code is running. For me to easily see, I'd like it to run as a toast notification that will make it very obvious that code is running. 

**Proposed technical implementation details (optional)**

Running powershell code will popup toast notification when running script. 
Options: 
- progress: Autodismiss after completion after designated number of secs
- Build in way that could provide enhanced progress outputting such as a progress bar. (I've seen gist snippet on doing this manually). 
- If error, provide the basic error message and don't autodismiss.
- Option to disable and just use status bar "running" icon that currently exists if don't want to opt in to this.

**Comment by rkeithhill** [comment]
There **is** an indicator for when PowerShell is executing script. It is in the status bar:

![image](https://user-images.githubusercontent.com/5177512/54661924-a7633e80-4aa1-11e9-8a7f-0437e90624e5.png)

**Comment by sheldonhull** [comment]
I agree it would be annoying if done improperly. I'm definitely recommending it autodismisses, or after delay removes itself. I saw this down with @SeeminglyScience doing something EditorServices or some snippet before and it was really cool. It provided a running progress bar so you could see the status at a glance. 

The key is that it shouldn't require clicks to deal with it, just be better information on something running.

**Comment by SeeminglyScience** [comment]
@sheldonhull I don't recall doing anything *useful* with status.  I think at some point I posted a no-op demo like this:

```powershell
for ($i = 1; $i -lt 11; $i++) {
    $psEditor.Window.SetStatusBarMessage(
        '' +
            [char]::ConvertFromUtf32(0x2595) +
            ([char]::ConvertFromUtf32(0x2589).ToString() * $i).
                PadRight(10, [char]::ConvertFromUtf32(0x2594)) +
            [char]::ConvertFromUtf32(0x258f),
        <# timeout: #> 600)

    Start-Sleep -Milliseconds 500
}
```

But that's about it. Really though, any useful graphic showing progress would have to be written into whatever script is running.  We can't really infer how many more steps/how much time/etc is left in a script.

**Comment by sheldonhull** [comment]
@SeeminglyScience i'll try and track it down. I think you or someone else perhaps then gave a snippet for write progress using the toast. I was thinking how useful it would be as a general "running" status in the bottom.
